### PR TITLE
chore: add default breakpointhook and styling

### PR DESCRIPTION
**Pull request by felipechiave**
*Issue #, if available:*

*Description of changes:*
On components with variants:
- Added generation of default breakpoint map and integrated into variants
- Added transition style

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts [hunk 1]
```diff
@@ -594,7 +594,7 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
       factory.createBindingElement(
         factory.createToken(ts.SyntaxKind.DotDotDotToken),
         undefined,
-        factory.createIdentifier('rest'),
+        factory.createIdentifier(hasVariant ? 'restProp' : 'rest'),
```
Comment (Jshhhh) [inline packages/codegen-ui-react/lib/react-studio-template-renderer.ts:597]: What is the purpose of using 'restProp' vs 'rest'?

Comment (felipechiave) [reply packages/codegen-ui-react/lib/react-studio-template-renderer.ts:597]: If a component has variants, we want to include a default transition style on the parent component. In a component-with-variants case, using restProp initially lets us define rest with a default transition style later (see line 778 in react-studio-template-renderer.ts)

**Review by Jshhhh — APPROVED** [review]
Nice work! LGTM

File: packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react.test.ts [hunk 2]
```diff
@@ -44,6 +44,12 @@ describe('amplify render tests', () => {
     it('should generate a simple image component', () => {});
 
     it('should generate a simple string component', () => {});
+
+    it('should generate a simple component without variant specific generation', () => {
+      const generatedCode = generateWithAmplifyRenderer('buttonGolden');
+      expect(generatedCode.componentText.indexOf('restProp')).toEqual(-1);
```
Comment (letsbelopez) [inline packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react.test.ts:50]: I think string.contains would be a better method for tests. could read pretty natural like this I think.

expect(generatedCode.componentText.contains('restProp')).toBe(false);

File: packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react.test.ts [hunk 3]
```diff
@@ -232,6 +238,12 @@ describe('amplify render tests', () => {
       const generatedCode = generateWithAmplifyRenderer('componentWithObjectVariants');
       expect(generatedCode).toMatchSnapshot();
     });
+
+    it('should have variant specific generation', () => {
+      const generatedCode = generateWithAmplifyRenderer('componentWithVariants');
+      expect(generatedCode.componentText.indexOf('restProp')).toBeGreaterThanOrEqual(0);
+      expect(generatedCode.componentText.indexOf('breakpointHook')).toBeGreaterThanOrEqual(0);
```
Comment (letsbelopez) [inline packages/codegen-ui-react/lib/__tests__/studio-ui-codegen-react.test.ts:245]: same as above comment

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts [hunk 4]
```diff
@@ -719,10 +721,103 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
...
+   *     const breakpointHook = useBreakpointValue({
+   *        base: 'base',
+   *        large: 'large',
+   *        medium: 'medium',
+   *        small: 'small',
+   *        xl: 'xl',
+   *        xxl: 'xxl',
+   *     });
+   */
+  private buildDefaultBreakpointMap() {
```
Comment (letsbelopez) [inline packages/codegen-ui-react/lib/react-studio-template-renderer.ts:734]: Is it useful to make this into a helper function? A function that creates a hook maybe?

createHookStatement('breakpointHook', 'useBreakpointValue', createObjectLiteralHelper({...base: 'base'}));

**Review by letsbelopez — CHANGES_REQUESTED** [review]
One question and one request in the tests

File: packages/codegen-ui-react/lib/react-studio-template-renderer.ts (hunk 4 above)
Comment (felipechiave) [reply packages/codegen-ui-react/lib/react-studio-template-renderer.ts:735]: Could be, especially if we want to add additional hooks to components in the future, added.

**Review by letsbelopez — APPROVED** [review]

**Pull request merged**